hazelcast/src/hazelcast/client/serialization.cpp

```diff
@@ -771,6 +806,42 @@ data_output::write(const hazelcast_json_value& value)
     write<std::string>(value.to_string());
 }
 
+void
+data_output::check_available(size_t index, int requested_length)
+{
+    if (index < 0) {
+        BOOST_THROW_EXCEPTION(exception::illegal_argument(
+          "DataOutput::checkAvailable",
+          (boost::format("Negative pos! -> %1%") % index).str()));
+    }
+
+    size_t available = output_stream_.size() - index;
+
+    if (requested_length > (int)available) {
+        BOOST_THROW_EXCEPTION(exception::illegal_argument(
+          "DataOutput::checkAvailable",
+          (boost::format("Cannot write %1% bytes!") % requested_length).str()));
+    }
+}
+
+void
+data_output::write_boolean_bit_at(size_t index,
+                                  size_t offset_in_bits,
+                                  bool value)
+{
+    if (is_no_write_) {
+        return;
+    }
+    check_available(index, 1);
```

Comment:
It checks if the existing one exists. If you are comparing against java, there all array accesses are checked and throws IndexOutofBound if there is space. In cpp, you need to be more careful because there is no such check by default.

Reply:
Ok but you removed the line. Did you do that by accident?